collector/src/main/java/com/bitmovin/analytics/features/segmenttracking/SegmentTrackingFeature.kt

```diff
@@ -0,0 +1,50 @@
+package com.bitmovin.analytics.features.segmenttracking
+
+import com.bitmovin.analytics.data.AdEventData
+import com.bitmovin.analytics.data.EventData
+import com.bitmovin.analytics.features.EventSource
+import com.bitmovin.analytics.features.Feature
+import com.bitmovin.analytics.stateMachines.PlayerEvent
+import com.bitmovin.analytics.stateMachines.PlayerState
+import java.util.*
+
+class SegmentTrackingFeature(private vararg val eventSources: EventSource<SegmentTrackingEventListener>) : Feature<SegmentTrackingFeatureConfig>(), SegmentTrackingEventListener {
```

Comment:
we might want to drop the `Feature` part as I see us structuring most of the new code around these "feature" buckets anyways.. so a SegmentTracker that then gets used by multiple other things again .. 🤔